infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationSourceManager.java

```diff
@@ -183,7 +169,7 @@ public void shutdown() {
 
         log.info("Shutdown Log Replication.");
         logReplicationFSM.shutdown();
-        runtime.shutdown();
+        isShutdown = true;
```

Comment:
the runtime will now get shutdown here only - https://github.com/CorfuDB/CorfuDB/blob/log_replication_v2/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L226 ?

Reply:
replying to both the comments about runtime here:
for https://github.com/CorfuDB/CorfuDB/pull/3731#discussion_r1277821278 : No. Since the sessions will be sharing the runtime, we will not be shutting the runtime upon an FSM shutdown.
So the runtime would only be shut down at the shutdown of discoveryService. I had missed that, will push the fix soon